app-typescript/components/SelectGrid.tsx

```diff
@@ -19,17 +19,21 @@ interface IProps {
     getItems(searchString: string | null): Promise<Array<IItem>>;
     onChange(value: IItem): void;
     itemTemplate: React.ComponentType<{ item: IItem | null }>;
-    triggerTemplate: React.ComponentType<any>;
+    triggerTemplate: React.ComponentType<{onClick: (e: React.SyntheticEvent) => void; }>;
     label: string;
     filterPlaceholder?: string;
 }
 
 interface IState {
     items: Array<IItem>;
     loading: boolean;
-    isOpen: boolean;
 }
 
+const VERTICAL_OFFSET = 3;
+const HORIZONTAL_OFFSET = 1;
+const PAGE_OFFSET = 15;
+const FIRST_ROW_INDEXES = [0, 1, 2];
```

Comment:
I wonder if it would be more readable to work with something like:
```
const GRID_COLS = 3;
const GRID_ROWS = 5;
```
or
```
const GRID_WIDTH = 3;
const GRID_HEIGHT = 5;
```
it's not clear why there would be any offsets